### PR TITLE
Inline Help: Add dismiss button to Checklist Prompt task

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* This PR is intended to be merged into #29409 
* Adds a Dismiss button which closes the Inline Help popup next to the action button.
* Sets the checklist prompt global variable to false on Dismiss.
* TODO: Can we/should we remove the query string?

screen shot 2018-12-20 at 9 58 34 am

#### Testing instructions

* Ensure you have both #29409 and this PR checked out
* Navigate to any site that is eligible for the checklist and add `?checklist` to the URL.
* Open the Inline Help pop-up and you should see the checklist prompt with a Dismiss button.
* Check to make sure the button dismisses the prompt.

### CODE REVIEW

**Comment by mattwiebe** [comment]
@sixhours this PR is attempting to merge `add/inline-help-first-task` into `add/inline-help-checklist-dismiss-button` - we will want it to merge into `master` to take effect. Can you start a new PR?

**Comment by adamziel** [comment]
I rebased this one after #29409 got merged - can I get a second pair of eyes on this before merging please?

File: client/my-sites/checklist/wpcom-checklist/checklist-prompt/task.jsx [hunk 1]
```diff
@@ -20,6 +23,19 @@ class ChecklistPromptTask extends PureComponent {
 		onClick: PropTypes.func,
 		title: PropTypes.string.isRequired,
 		translate: PropTypes.func.isRequired,
+		closePopover: PropTypes.func.isRequired,
+	};
+
+	dismissPopup = () => {
+		//Remove query string, too
```
Comment (southp) [inline client/my-sites/checklist/wpcom-checklist/checklist-prompt/task.jsx]: I don't see this happening on my end. Is this something missed in the current implementation?

**Comment by southp** [comment]
In general the code LGTM and the test plan works. I do have an inquiry left as a inline comment, though. If it's not relevant, please feel free to ignore and merge this.

**Comment by adamziel** [comment]
While it would be a nice addition, it's not required for now so I just removed that comment and I'm going to go ahead and merge it. Thank you so much for reviewing!

**Pull request merged**